### PR TITLE
Render text in render system

### PR DESCRIPTION
**Pull request by averhagen**
The ScoreScreen used to make its own font then handle the rendering of that font. This branch takes the font generation and moves it into a new Assets class. The Assets class has a public AssetManager that it holds the assets in. The Assets class needs updated to include all the game assets. This branch also took the rendering of fonts and added them to the RenderSystem.

### CODE REVIEW

File: core/src/com/goonsquad/galactictd/managers/Assets.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public void loadAssets() {
+        loadLoadingScreenAssets();
+        loadNixieFont();
+    }
+
+    private void loadLoadingScreenAssets() {
+        //Put assets here that are needed on loading screen.
+        //Calling finish loading will insure all the needed assets are loaded.
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/managers/Assets.java]: > insure

ensure?

File: core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java [hunk 2]
```diff
@@ -65,23 +67,29 @@ protected void begin() {
...
+                        entityPosition.rotation,
+                        0, 0,
+                        entityRenderable.texture.getWidth(),
+                        entityRenderable.texture.getHeight(),
+                        false, false);
+            }
+            if (textComponentMapper.has(entityId)) {
+                Text text = textComponentMapper.get(entityId);
+                text.font.draw(batch, text.text, entityPosition.x, entityPosition.y);
+            }
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/systems/graphics/RenderSystem.java:92]: Might be worth looking into how we could generalize this into a single call that would delegate the actual drawing interactions to a subclass. Or if we want to keep it like this, can an entity be both a Renderable and a Text (based on the Constructor, I'd say no)?

File: core/src/com/goonsquad/galactictd/managers/Assets.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    private void loadNixieFont() {
+        FileHandleResolver resolver = new InternalFileHandleResolver();
+        manager.setLoader(FreeTypeFontGenerator.class, new FreeTypeFontGeneratorLoader(resolver));
+        manager.setLoader(BitmapFont.class, ".ttf", new FreetypeFontLoader(resolver));
+        FreetypeFontLoader.FreeTypeFontLoaderParameter loaderParams = new FreetypeFontLoader.FreeTypeFontLoaderParameter();
+        loaderParams.fontFileName = "NixieOne.ttf";
+        loaderParams.fontParameters.size = 48;
+        loaderParams.fontParameters.borderColor = Color.WHITE;
+        loaderParams.fontParameters.borderWidth = 1.3f;
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/managers/Assets.java]: Feels like some magic numbers in here, not sure if we need to do anything about that.

File: core/src/com/goonsquad/galactictd/managers/Assets.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        this.manager = new AssetManager();
+    }
+
+    public void loadAssets() {
+        loadLoadingScreenAssets();
+        loadNixieFont();
+    }
+
+    private void loadLoadingScreenAssets() {
+        //Put assets here that are needed on loading screen.
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/managers/Assets.java]: Should this be done in this commit? If not, make this a TODO

File: core/src/com/goonsquad/galactictd/systems/initialization/ScoreScreenInitSystem.java [hunk 5]
```diff
@@ -117,4 +124,29 @@ private int createGreenShip() {
...
+            label = archetypeBuilder.buildArchetype(ScoreScreenArchetypeBuilder.textLabel);
+            labelText = textComponentMapper.get(label);
+            labelText.text = "0";
+            labelText.font = gameInstance.assets.manager.get("nixie48.ttf", BitmapFont.class);
+
+            labelPosition = positionComponentMapper.get(label);
+            labelPosition.x = textPosition.x * 3;
+            labelPosition.y = textPosition.y * textHeightFactor;
+            textHeightFactor -= 2;
+        }
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/systems/initialization/ScoreScreenInitSystem.java]: Some explanatory comments about these magic numbers would be great (consider using constants or making them configurable as well).

File: core/src/com/goonsquad/galactictd/managers/Assets.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+public class Assets implements Disposable {
+    public AssetManager manager;
+
+    public Assets() {
+        this.manager = new AssetManager();
+    }
+
+    public void loadAssets() {
+        loadLoadingScreenAssets();
+        loadNixieFont();
```
Comment (kevinfulkerson) [inline core/src/com/goonsquad/galactictd/managers/Assets.java]: Consider the order of calls in this method: do we want to say we are done loading if we have yet to load fonts?

**Comment by averhagen** [comment]
Like the thoughts. Some of the questions might be answered by looking at the [asset](https://github.com/kevinfulkerson/towerdefense/tree/move-asset-manager-to-new-class) branch. I like the text being in the same system as the render right now because it allows text to be draw according to layer along with the other renderables. Maybe some sort of weaving of two systems could be done to there. As of now an entity could have a renderable and a texture and they would both be rendered.

**Comment by averhagen** [comment]
I've cleaned up some of the code in ScoreScreenInitSystem.java and Assets.java. I've held off on refactoring the render system for now but it could be a place for changes in the future.

**Comment by kevinfulkerson** [comment]
Okay, looks good to me.

**Pull request merged**